ios/Classes/NativeBridge/Handlers/ResourcesHandler.swift

```diff
@@ -34,226 +30,86 @@ class ResourcesHandler: FetchResourcesHandlerProtocol {
-            return ONGParametersEncoding.JSON
-        case "application/x-www-form-urlencoded":
-            return ONGParametersEncoding.formURL
-        default:
-            return ONGParametersEncoding.JSON
-        }
-    }
-
-    //MARK: - Bridge
-    func fetchAnonymousResource(_ path: String, parameters: [String: Any?], completion: @escaping FlutterResult) {
-        Logger.log("fetchAnonymousResource", sender: self)
-        
-        let newParameters = generateParameters(from: parameters, path: path)
-
-        OneginiModuleSwift.sharedInstance.resourceRequest(isImplicit: false, parameters: newParameters) { (data, error) in
-            if let _errorResource = error {
-                completion(_errorResource)
-                return
-            } else {
-                completion(data)
-            }
-        }
+private extension ResourcesHandler {
+    func generateONGResourceRequest(_ details: OWRequestDetails) -> ONGResourceRequest {
+        Logger.log("generateONGResourceRequest", sender: self)
+        return ONGResourceRequest(path: details.path,
+                                  method: details.method.stringValue,
+                                  body: details.body?.data(using: .utf8),
+                                  headers: getRequestHeaders(details.headers)
+                                 )
     }
 
-    func fetchSimpleResources(_ path: String, parameters: [String: Any?], completion: @escaping FlutterResult) {
-        Logger.log("fetchSimpleResources", sender: self)
-        let newParameters = generateParameters(from: parameters, path: path)
-        OneginiModuleSwift.sharedInstance.resourceRequest(isImplicit: false, isAnonymousCall: false, parameters: newParameters) { (_data, error) in
-            if let _errorResource = error {
-                completion(_errorResource)
-                return
-            } else {
-                completion(_data)
-            }
-        }
-    }
-
-    func fetchResourceWithImplicitResource(_ path: String, parameters: [String: Any?], completion: @escaping FlutterResult) {
-        Logger.log("fetchResourceWithImplicitResource", sender: self)
-        guard let _ = ONGUserClient.sharedInstance().implicitlyAuthenticatedUserProfile() else {
-            completion(SdkError(.unauthenticatedImplicitly).flutterError())
-            return
-        }
+    func getRequestHeaders(_ headers: [String?: String?]?) -> [String: String]? {
+        Logger.log("getRequestHeaders", sender: self)
+        guard let headers = headers else { return nil }
 
-        let newParameters = generateParameters(from: parameters, path: path)
-        let scopes = newParameters["scope"] as? [String]
-        
-        OneginiModuleSwift.sharedInstance.resourceRequest(isImplicit: true, parameters: newParameters) { (data, error) in
-            completion(data ?? error)
-        }
+        return headers.filter { $0.key != nil && $0.value != nil } as? [String: String] ?? [:]
```

Comment:
Either remove `?? [:]` or mark return value as non optional `-> [String: String]`. Now you can have 2 cases: empty dict or nil - the receiver would need to check both cases/unwrap. If possible go for non-optional option.

Reply:
removed ?? [:] case